lib/kube/proxy/forwarder.go

```diff
@@ -493,6 +507,42 @@ func (f *Forwarder) setupContext(ctx auth.Context, req *http.Request, isRemoteUs
 	return authCtx, nil
 }
 
+func (f *Forwarder) authorize(ctx context.Context, actx *authContext) error {
+	if actx.teleportCluster.isRemote {
+		return nil
+	}
+	if actx.kubeCluster == "" {
+		// This should only happen for remote clusters (filtered above), but
+		// check and report anyway.
+		f.WithField("auth_context", actx.String()).Debug("skipping authorization due to unknown kubernetes cluster name")
+		return nil
+	}
+	servers, err := f.AccessPoint.GetKubeServices(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Check authz against the first match.
+	//
+	// We assume that users won't register two identically-named clusters with
+	// mis-matched labels. If they do, expect weirdness.
+	for _, s := range servers {
+		for _, ks := range s.GetKubernetesClusters() {
+			if ks.Name != actx.kubeCluster {
+				continue
+			}
+			if err := actx.Checker.CheckAccessToKubernetes(defaults.Namespace, ks); err != nil {
```

Comment:
I know namespaces aren't really used in Teleport, but you used `server.GetNamespace()` above so I'd try to stay consistent.

Reply:
done